src/CMakeLists.txt

```diff
@@ -185,8 +185,15 @@ add_compile_options(
   $<$<CXX_COMPILER_ID:Clang>:-Wno-gnu-zero-variadic-macro-arguments>
   # Apple clang 14.0.0 deprecates sprintf, which generates 900 warnings.
   $<$<CXX_COMPILER_ID:AppleClang>:-Wno-deprecated-declarations>
+  $<$<BOOL:${ASAN}>:-fsanitize=address>
+  $<$<BOOL:${ASAN}>:-fno-omit-frame-pointer>
+  $<$<BOOL:${ASAN}>:-g>
 )
 
+if (ASAN)
+  add_link_options(-fsanitize=address)
```

Comment:
Are both of the add_link_options required? What's the difference between this one and the one below on line 288?

Reply:
I believe only add_compile_options and add_link_options are required. These commands modify the COMPILE_OPTIONS and LINK_OPTIONS properties for the directory, so the options will be added for all executables and libraries in the current directory and below _that are added after the command_. I was confused about this functionality and thought target_compile_options and target_link_options would be required as well.

I have removed the target_compile_options and target_link_options, and I am rebuilding to make sure it still works.

Reply:
I only dislike a rebase after an initial review as it breaks the show changes since last review. As this hasn't been reviewed yet it is fine to rebase before the initial review (at least by me)

Reply:
I removed target_compile_options and target_link_options and rebuilt both Clang and GNU. Address sanitizer still produces the same output. Now only add_compile_options and add_link_options are used.